source/Mlos.Agent.Server/MlosAgentServer.cs

```diff
@@ -220,6 +232,71 @@ public static void Main(string[] args)
             Console.WriteLine("Mlos.Agent exited.");
         }
 
+        private static void ShowUsageHelp<T>(ParserResult<T> parserResult, IEnumerable<Error> errors = null, string msg = null)
+        {
+            if (msg != null)
+            {
+                Console.Error.WriteLine(msg);
+                Console.Error.WriteLine(string.Empty);
+            }
+
+            if (errors == null)
+            {
+                errors = new List<Error>();
+            }
+
+            HelpText helpText = null;
+            if (errors.IsVersion())
+            {
+                helpText = HelpText.AutoBuild(parserResult);
+            }
+            else
+            {
+                helpText = HelpText.AutoBuild(
+                    parserResult,
+                    onError: ht =>
+                    {
+                        return HelpText.DefaultParsingErrorsHandler(parserResult, ht);
+                    },
+                    e => e);
+                helpText.AddNewLineBetweenHelpSections = true;
+                helpText.AddPreOptionsLines(new[]
+                    {
+                        // Use a single long line of text to let the help output get wrapped automatically for us.

```

Comment:
Could you move help and cmd line parsing to new file ?

Reply:
Is it necessary?  I kinda like having the description of what the main code is doing in that file.  Makes it easier for a newbie to find/discover/understand at a glance.

Reply:
Yes, please .Having one big file does not make things easier.

Reply:
Kk, will reorg

Reply:
Done.  Let me know if you have any other feedback on it.